src/lib/containers/entity/page/title_bar/EntityPageTitleBar.tsx

```diff
@@ -0,0 +1,106 @@
+  entityId: string
+  versionNumber?: number
+  /* Props that will be passed through to an EntityActionMenu component */
+  entityActionMenuProps?: EntityActionMenuProps
+  /* Callback invoked when an ACT member clicks "Add Conditions for Use". */
+  onActMemberClickAddConditionsForUse: () => void
+}
+
+/**
+ * The EntityPageTitleBar component is shown on every Entity page in Synapse, and shows an Entity's metadata and
+ * provides controls to view and manipulate the Entity.
+ */
+export default function EntityPageTitleBar(props: EntityPageTitleBarProps) {
+  const { entityId, versionNumber, entityActionMenuProps } = props
+
+  const toggleShowVersionHistory =
+    entityActionMenuProps?.actionConfiguration['SHOW_VERSION_HISTORY']?.onClick
+
+  const { data: bundle, isLoading } = useGetEntityBundle(
+    entityId,
+    versionNumber,
+  )
+
+  const name =
+    bundle?.entity && isDockerRepository(bundle.entity)
+      ? bundle.entity.repositoryName
+      : bundle?.entity?.name
+
+  return (
+    <div>
+      <Box
+        sx={{
+          padding: '30px 40px',
+          backgroundColor: 'grey.100',
+        }}
+      >
+        <Stack direction={'row'}>
+          <Stack
+            direction={'row'}
+            alignItems={'center'}
+            sx={{
+              flexGrow: 1,
+            }}
+          >
+            <ConditionalWrapper
+              condition={isLoading}
+              wrapper={Skeleton}
+              wrapperProps={{
+                variant: 'circular',
+              }}
+            >
+              <Avatar sx={{ bgcolor: 'grey.300', height: 48, width: 48 }}>
+                {bundle?.entityType && (
+                  <IconSvg
+                    icon={
+                      bundle?.entityType
+                        ? type2SvgIconName[bundle.entityType]
+                        : 'file'
+                    }
+                    wrap={false}
+                    sx={{ color: 'grey.700' }}
+                  />
+                )}
+              </Avatar>
+            </ConditionalWrapper>
+            <Box sx={{ marginLeft: '10px', flexGrow: 1 }}>
+              <Stack direction={'row'} alignItems={'center'} gap={'3px'}>
+                <ConditionalWrapper condition={isLoading} wrapper={Skeleton}>
+                  <Typography variant={'headline2'}>
+                    {name ?? 'Placeholder Name'}
+                  </Typography>
+                </ConditionalWrapper>
+                {bundle?.entity?.id && (
+                  <FavoriteButton entityId={bundle.entity.id} />
+                )}
+              </Stack>
+              <EntityTitleBarVersionInfo
+                entityId={entityId}
+                versionNumber={versionNumber}
+                toggleShowVersionHistory={toggleShowVersionHistory}
+              />
+            </Box>
+          </Stack>
+          {entityActionMenuProps && (
+            <EntityActionMenu {...entityActionMenuProps} />
+          )}
```

Comment:
Render the action menu here instead of trying to do something hacky like React-in-GWT-in-React-in-GWT

Reply:
Out of curiosity, did you attempt such a hack (and if so, how far did you get)?

Reply:
Yes; I couldn't manage to attach a GWT widget to an arbitrary DOM element within a React component. Using [this SO post](https://stackoverflow.com/questions/6183181/how-to-add-a-custom-widget-to-an-element/6186394#6186394) as a reference, `HTMLPanel.wrap` threw an error when trying to use it.